DiscordChatExporter.Core/Exporting/Filtering/MentionsMessageFilter.cs

```diff
@@ -0,0 +1,20 @@
+﻿using DiscordChatExporter.Core.Discord.Data;
+using System.Linq;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public class MentionsMessageFilter : MessageFilter
+    {
+        private readonly string _value;
+
+        public MentionsMessageFilter(string value)
+        {
+            ///TODO: should we validate input here?
+            _value = value;
+        }
+
+        //match either a username + discriminator or an id
+        public override bool Filter(Message message) =>
+            message.MentionedUsers.Any(user => _value == user.FullName || _value == user.Id.ToString());
```

Comment:
Maybe we should also try to match just the `user.Name`? In case the user doesn't specify the discriminator. I know it will then be able to match multiple people with same names, but maybe that's also the user's intention in such case.

(Also the same applies to `from:` filter)

Reply:
Sounds good.